sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -238,7 +238,7 @@ class Dataset[T] private[sql](
   private[sql] def showString(
       _numRows: Int, truncate: Int = 20, vertical: Boolean = false): String = {
     val numRows = _numRows.max(0)
-    val takeResult = toDF().take(numRows + 1)
+    val takeResult = toDF().take(if (numRows == Int.MaxValue) numRows else numRows + 1)
```

Comment:
Normally, we split it to two lines. How about ?
```
    val numRows = _numRows.max(0).min(Int.MaxValue - 1)
    val takeResult = toDF().take(numRows + 1)
```

Reply:
yea, looks great. I updated.